test/addons-napi/7_factory_wrap/myobject.cc

```diff
@@ -0,0 +1,107 @@
+#include "myobject.h"
+
+MyObject::MyObject() : env_(nullptr), wrapper_(nullptr) {}
+
+MyObject::~MyObject() { napi_delete_reference(env_, wrapper_); }
+
+void MyObject::Destructor(void* nativeObject, void* /*finalize_hint*/) {
+  reinterpret_cast<MyObject*>(nativeObject)->~MyObject();
```

Comment:
This seems sketchy … why isn’t this a `delete`? Where is the memory freed that’s being allocated by the `new MyObject()` call?

(Also, re: all the `reinterpret_cast`ing… we do try to avoid that if a `static_cast` does the job just fine)